src/pages/Pools/helpers/index.ts

```diff
@@ -129,4 +132,75 @@ export const estimateShare = ({
   return share;
 };
 
-export { generateMsgFrontierAddToken, getInfoLiquidityPool, isBigIntZero };
+export const getStatisticData = (data: PoolTableData[]) => {
+  const statisticData = data.reduce(
+    (acc, curr) => {
+      acc.volume = acc.volume + toDisplay(curr.volume24Hour);
+      acc.totalLiquidity = acc.totalLiquidity + curr.totalLiquidity;
+
+      return acc;
+    },
+    {
+      volume: 0,
+      totalLiquidity: 0,
+      totalClaimable: 0
+    }
+  );
+
+  return statisticData;
+};
+
+export const getClaimableInfoByPool = ({ pool, totalRewardInfoData }) => {
+  const rewardPerSecInfoData = JSON.parse(pool.rewardPerSec);
+
+  const currentPoolReward = totalRewardInfoData?.reward_infos?.find((reward) =>
+    isEqual(reward.staking_token, pool.liquidityAddr)
+  );
+  const totalRewardAmount = BigInt(currentPoolReward?.pending_reward ?? 0);
+
+  // unit LP
+  const totalRewardPerSec = rewardPerSecInfoData.assets
+    .map((asset) => BigInt(asset.amount))
+    .reduce((a, b) => a + b, BigInt(0));
+
+  const results = rewardPerSecInfoData.assets
+    .filter((asset) => parseInt(asset.amount))
+    .map(async (asset) => {
+      const pendingWithdraw = BigInt(
+        currentPoolReward?.pending_withdraw.find((e) => isEqual(e.info, asset.info))?.amount ?? 0
+      );
+
+      const amount = (totalRewardAmount * BigInt(asset.amount)) / totalRewardPerSec + pendingWithdraw;
+
+      let token =
+        'token' in asset.info ? cw20TokenMap[asset.info.token.contract_addr] : tokenMap[asset.info.native_token.denom];
+
+      // only for atom/scatom pool
+      if (!token && 'token' in asset.info && asset.info?.token?.contract_addr) {
```

Comment:
Since we're using a lot of optional chaining, do we enable `strictNullChecks` with tsconfig? 

From a @tubackkhoa perspective, should we use `?.` or use more explicit check ?